src/Model/ModelManager.php

```diff
@@ -23,16 +23,27 @@
 use Sonata\DoctrineMongoDBAdminBundle\Datagrid\ProxyQuery;
 use Sonata\Exporter\Source\DoctrineODMQuerySourceIterator;
 use Symfony\Bridge\Doctrine\ManagerRegistry;
-use Symfony\Component\Form\Exception\PropertyAccessDeniedException;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
 
 class ModelManager implements ModelManagerInterface
 {
     public const ID_SEPARATOR = '-';
+
+    /**
+     * @var ManagerRegistry
+     */
     protected $registry;
 
-    public function __construct(ManagerRegistry $registry)
+    /**
+     * @var PropertyAccessorInterface
+     */
+    private $propertyAccessor;
+
+    public function __construct(ManagerRegistry $registry, ?PropertyAccessorInterface $propertyAccessor = null)
```

Comment:
You could add the injection in https://github.com/sonata-project/SonataDoctrineMongoDBAdminBundle/blob/3.x/src/Resources/config/doctrine_mongodb.xml#L4

Reply:
And what about requiring the param in the next major ?  (And add a deprecation in 3.x if it's not provided)